### ISSUE TITLE
Implement support for dynamic scaffolding which was removed in Grails 3.0

### ISSUE THREAD

**Issue by smitp33** [grails/grails-core#681]
Just tried the latest version and using a basic domain class with the following in the controller:

static scaffold = true

When calling the controller with the default action it tries to send me to the index page and fails as no index is present.

Tried the same on V 2.5.0 and all is fine.

OS is Windows 7 
JDK is 1.7.0.79

Noted that there was an issue raised on the scaffolding plugin not working in 3.0 M1 release with almost exactly the same behavior as I have above.

**Comment by zyro23** [comment]
see https://groups.google.com/d/topic/grails-dev-discuss/6R2YaF96Uts/discussion

**Comment by smitp33** [comment]
Thanks. 

Why not update the documentation to or the other defect with this info to save other people wasting time working out if the were making a mistake 😀

**Comment by eugene-kamenev** [comment]
@smitp33 you can help with docs yourself by providing pull request

**Comment by jeffbrown** [comment]
> Scaffolding plugin broken in 3.0.1

I don't think it is broken.  There is ongoing discussion about whether or not the dynamic scaffolding feature should be resurrected but it was removed intentionally so it is currently behaving as designed with respect to this.  The user guide was updated a while back with the change at https://github.com/grails/grails-doc/commit/d9a55891c40e19b589c3f1030530341872941d09.  It was an oversight that the dynamic scaffolding docs were left in the user guide.

**Comment by charlesw** [comment]
Has support for the generate-views command been removed from scaffolding in 3.0.1? I've been trying to follow an older [tutorial](http://www.vogella.com/tutorials/Grails/article.html) and all these undocumented changes are really throwing a spanner in the works.

**Comment by jeffbrown** [comment]
> Has support for the generate-views command been removed from scaffolding in 3.0.1?

Yes.

**Comment by ikakara** [comment]
Question: why is this a 2.5.1 milestone?  Shouldn't this be a 3.x milestone?

**Comment by anshbansal** [comment]
Was dynamic scaffolding added back?

**Comment by graemerocher** [comment]
@anshbansal yup

**Comment by anshbansal** [comment]
@graemerocher So anything is different from before?

Because I added in docs that dynamic scaffolding was removed in the upgrade guide. Also @jeffbrown removed most of it.

**Comment by graemerocher** [comment]
@anshbansal no, works the same as before

**Comment by jeffbrown** [comment]
The docs should be restored before the release.

**Comment by anshbansal** [comment]
@jeffbrown I'll send a PR in a 1-2 days.

**Comment by jeffbrown** [comment]
@anshbansal I was almost certain you would. ;)

Thanks for the help.

**Comment by anshbansal** [comment]
@jeffbrown You can always help me back by adding something in https://github.com/grails/grails-core/issues/9088. Then I can try and help with the code also ;)

**Comment by jeffbrown** [comment]
@anshbansal I added content to a number of the README.md files.  Not a giant priority right now, I am working on the last 2.5.1 issues.

**Comment by anshbansal** [comment]
@jeffbrown I understand.

**Comment by anshbansal** [comment]
I added the PR https://github.com/grails/grails-doc/pull/344 but looking at the code it seems that there are some differences.

- generate-views is not present as far as I can tell.
- Also I added back the docs for install-templates command. But looking at the code I can say that there are some differences in its functionality. Earlier it installed tempaltes for all code generation as per the docs. Now it will just copy the ones for scaffolding.
